### PR TITLE
Synchronize task threading

### LINKED ISSUE #930: Threading issue in LottieTask
Crash
```
java.lang.NullPointerException: Attempt to invoke virtual method 'void java.lang.Thread.interrupt()' on a null object reference
        at com.airbnb.lottie.LottieTask.stopTaskObserverIfNeeded(LottieTask.java:215)
        at com.airbnb.lottie.LottieTask.access$500(LottieTask.java:28)
        at com.airbnb.lottie.LottieTask$2.run(LottieTask.java:198)
```

Looks like on a single thread this can't happen as the thread is checked for null a few lines before:
https://github.com/airbnb/lottie-android/blame/master/lottie/src/main/java/com/airbnb/lottie/LottieTask.java#L216

So it seems that access to the thread isn't properly synchronized

### PR DESCRIPTION
**Pull request by gpeal**
fixes #930 

@elihart

### CODE REVIEW

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-04c40ab76fbcafd426241f64c9fe8d88b0daaa06-android26&a=27

File: lottie/src/main/java/com/airbnb/lottie/LottieTask.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ private void setResult(@Nullable LottieResult<T> result) {
    * a listener if neccesary.
    * @return the task for call chaining.
    */
-  public LottieTask<T> removeListener(LottieListener<T> listener) {
+  public synchronized LottieTask<T> removeListener(LottieListener<T> listener) {
     synchronized (successListeners) {
```
Comment (elihart) [inline lottie/src/main/java/com/airbnb/lottie/LottieTask.java]: can you remove `synchronized (successListeners)` now?

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-61d2741a94cfb520e4045be6a9c2d61040d18c07-android26&a=27

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-8ddbb50ce11dea01407cecf34cb6aaaccfde70d9-android26&a=27

File: lottie/src/main/java/com/airbnb/lottie/LottieTask.java [hunk 2]
```diff
@@ -154,7 +154,7 @@ private void notifyListeners() {
   }
 
   private void notifySuccessListeners(T value) {
-    // Allow listeners to remove themself in onResult.
+    // Allow listeners to remove them self in onResult.
```
Comment (tylerbwong) [inline lottie/src/main/java/com/airbnb/lottie/LottieTask.java]: "Allow listeners to remove themselves in onResult."

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-c22e362313ff2a3fc1ca16e85b305952f4c85038-android26&a=27

**Comment by gpeal** [comment]
Snapshot Test Results: https://happo.io/compare?q=lottie-master-android26..lottie-c22d3384f166b9f830241a28ac68dc06d6c77c57-android26&a=27

**Pull request merged**